apple/MarkdownParser.mm

```diff
@@ -0,0 +1,68 @@
+#import "MarkdownParser.h"
+#import <RNLiveMarkdown/MarkdownGlobal.h>
+#import <React/RCTLog.h>
+#import <mutex>
+
+@implementation MarkdownParser {
+  NSString *_prevText;
+  NSNumber *_prevParserId;
+  NSArray<MarkdownRange *> *_prevMarkdownRanges;
+}
+
+- (NSArray<MarkdownRange *> *)parse:(NSString *)text withParserId:(nonnull NSNumber *)parserId {
+  @synchronized (self) {
```

Comment:
Is `@synchronized` needed here? Or do you just want to be explicit about the behavior?

Reply:
`MarkdownParser` can be used from multiple threads (i.e. UI and JS thread) and we read/write `_prev*` fields so that's why I added `@synchronized`.

Reply:
Yep, moving the lock would make `@synchronized` unnecessary. But the lock itself is also subject to change as I'd like to eliminate keeping worklet runtime in global at some point so probably the code in the current shape is more future-proof.